code/perception/src/traffic_light_detection/src/traffic_light_detection/traffic_light_training.py

```diff
@@ -0,0 +1,151 @@
+    load_image
+from data_generation.weights_organizer import WeightsOrganizer
+from traffic_light_detection.classification_model import ClassificationModel
+from torchvision.transforms import ToTensor
+from traffic_light_config import TrafficLightConfig
+
+
+def parse_args():
+    """
+    Parses arguments for execution given by the command line.
+    @return: Parsed arguments
+    """
+    parser = argparse.ArgumentParser(description='Train traffic light network')
+    parser.add_argument('--epochs',
+                        help='number of epochs',
+                        type=int)
+    parser.add_argument('--num_saves',
+                        help='number of model-weights to be saved',
+                        type=int)
+    return parser.parse_args()
+
+
+class TrafficLightTraining:
+
+    def __init__(self, cfg):
+        """
+        Initializes an instance to train a traffic light classification model.
+        @param cfg: Config file for traffic light classification
+        """
+        self.cfg = cfg
+        train_transforms = t.Compose([
+            ToTensor(),
+            ResizeAndPadToSquare([32, 32]),
+            Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
+            # ApplyMask(dataset_root + "/mask.png")
+        ])
+        val_transforms = t.Compose([
+            ToTensor(),
+            ResizeAndPadToSquare([32, 32]),
+            Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
+        ])
+        self.train_dataset = ImageFolder(root=self.cfg.DATASET_PATH + "/train",
+                                         transform=train_transforms,
+                                         loader=load_image)
+        self.train_loader = DataLoader(dataset=self.train_dataset,
+                                       batch_size=self.cfg.BATCH_SIZE,
+                                       num_workers=self.cfg.NUM_WORKERS,
+                                       shuffle=True)
+        self.val_dataset = ImageFolder(root=self.cfg.DATASET_PATH + "/val",
+                                       transform=val_transforms,
+                                       loader=load_image)
+        self.val_loader = DataLoader(dataset=self.val_dataset,
+                                     batch_size=self.cfg.BATCH_SIZE,
+                                     num_workers=self.cfg.NUM_WORKERS)
+        self.model = ClassificationModel(num_classes=self.cfg.NUM_CLASSES,
+                                         in_channels=self.cfg.NUM_CHANNELS)
+        self.optimizer = Adam(self.model.parameters())
+        self.loss_function = torch.nn.CrossEntropyLoss()
+        self.weights_organizer = WeightsOrganizer(cfg=self.cfg,
+                                                  model=self.model)
+
+    def run(self):
+        """
+        Trains the model for a given amount of epochs
+        """
+        self.model.to(self.cfg.DEVICE)
+        tepoch = tqdm(range(self.cfg.EPOCHS))
+        for i in range(self.cfg.EPOCHS):
```

Comment:
would it make sense to use https://pytorch.org/docs/1.3.0/optim.html#torch.optim.lr_scheduler.ExponentialLR instead of your own implementation?

Reply:
Oh yes, forgot it 🙈

Reply:
Resolved in [cb23b91](https://github.com/ll7/paf22/pull/116/commits/cb23b91f07e462d9e67e01d33f34431376feedfe)